src/pubsub.ts

```diff
@@ -543,30 +543,60 @@ export class PubSub {
   }
   /**
    * Determine the appropriate endpoint to use for API requests, first trying
-   * the local `apiEndpoint` parameter. If the `apiEndpoint` parameter is null
-   * we try Pub/Sub emulator environment variable (PUBSUB_EMULATOR_HOST),
-   * otherwise the default JSON API.
+   * the local `apiEndpoint` parameter. If the `apiEndpoint` parameter is null,
+   * we try the standard `gcloud alpha pubsub` environment variable
+   * (CLOUDSDK_API_ENDPOINT_OVERRIDES_PUBSUB), and failing that, we try the
+   * Pub/Sub emulator environment variable (PUBSUB_EMULATOR_HOST).
+   * Otherwise the default production API is used.
    *
    * @private
    */
   determineBaseUrl_() {
-    const apiEndpoint = this.options.apiEndpoint;
-    if (!apiEndpoint && !process.env.PUBSUB_EMULATOR_HOST) {
+    // We allow an override from the client object options, or from
+    // one of these variables. The CLOUDSDK variable is provided for
+    // compatibility with the `gcloud alpha` utility.
+    const gcloudVarName = 'CLOUDSDK_API_ENDPOINT_OVERRIDES_PUBSUB';
+    const emulatorVarName = 'PUBSUB_EMULATOR_HOST';
+    const apiEndpoint =
+      this.options.apiEndpoint ||
+      process.env[gcloudVarName] ||
+      process.env[emulatorVarName];
```

Comment:
This could be backwards, but I would think "PUBSUB_EMULATOR_HOST" would be before "CLOUDSDK_API_ENDPOINT_OVERRIDES_PUBSUB" in precedence, as "CLOUDSDK" sounds like it's set automatically in that environment, compared to "PUBSUB_EMULATOR_HOST" which is deliberately set by the user. I suppose it's a rare scenario where both are defined, but we've always treated "PUBSUB_EMULATOR_HOST" as something a user would define as an alternative to "options.apiEndpoint".